src/app/redux/auth/authActions.js

```diff
@@ -30,36 +34,42 @@ const setErrorWrongPassword = () => {
   }
 }
 
-const signIn = (credentials) => {
-  const uri = process.env.REACT_APP_BASE_API_URL + '/users?q=' + credentials.email;
+const setError = (msg) => {
+  return {
+    type: actionTypes.AUTH_ERROR,
+    error: msg,
+  }
+}
 
+const signIn = (credentials) => {
   return (dispatch) => {
-    fetch(uri)
-    .then(res => res.json())
-    .then(data => {
-      if(data.length === 0) {
-        dispatch(setErrorUserNotExist());
-      } else {
-        if(credentials.password !== data[0].password) {
-        dispatch(setErrorWrongPassword());
-        } else {
-          dispatch(setActiveUser({...data[0]}));
+    dispatch(setError(null));
+    signInWithEmailAndPassword(authConnection, credentials.email, credentials.password)
+      .then(cred => {
+        getUserByEmail(cred.user.email)
+          .then(userData => {
+            dispatch(setActiveUser({ ...userData }));
 
-          localStorage.setItem("userEmail", data[0].email);
-          localStorage.setItem("userRole", data[0].role);
-          localStorage.setItem("userId", data[0].id);
-          localStorage.setItem("userName", data[0].name);
-        }
-      }
-    })
+            localStorage.setItem("userEmail", userData.email);
+            localStorage.setItem("userRole", userData.role);
+            localStorage.setItem("userId", userData.id);
+            localStorage.setItem("userName", userData.name);
+          })
+          .catch(error => {
+            dispatch(setError(error.message));
+          })
+      })
+      .catch(error => {
+        dispatch(setError(error.message));
+      })
```

Comment:
Is there a way to make this one cleaner?

Reply:
I shortened it by making the saving data to local storage part a separate function